[Gomock Generator] Panic if an interface contains an interface

For example:

```go
type Conn interface {
	Dialer
	BuildConnChain() string
	Ping() error
}
```

panic with:
```
panic: runtime error: index out of range

goroutine 55 [running]:
github.com/Scalingo/go-utils/gomock_generator/gomockgenerator.interfaceSignature(0xc0001be700, 0x3b, 0xc000018680, 0x4, 0xc0001dfe00, 0x83, 0x5c6f40, 0x3c)                                  
        /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/go-utils/gomock_generator/gomockgenerator/package.go:54 +0xc39                                                       
github.com/Scalingo/go-utils/gomock_generator/gomockgenerator.interfaceHash(0xc0001be700, 0x3b, 0xc000018680, 0x4, 0x2, 0xc0001be740, 0x40, 0x5a22e0)                                        
        /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/go-utils/gomock_generator/gomockgenerator/package.go:17 +0x67                                                        
github.com/Scalingo/go-utils/gomock_generator/gomockgenerator.generateMock(0x602260, 0xc0003f05a0, 0xc00001a270, 0x25, 0xc000018680, 0x4, 0xc00001a990, 0x2c, 0xc0001be700, 0x3b, ...)       
        /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/go-utils/gomock_generator/gomockgenerator/generate.go:173 +0x918                                                     
github.com/Scalingo/go-utils/gomock_generator/gomockgenerator.GenerateMocks.func1(0xc0000221c0, 0xc00000e6a0, 0xc00001a270, 0x25, 0xc0000e2000, 0x4f, 0x5e, 0xc00000c050, 0xc000058600, 0xc000018810, ...)
        /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/go-utils/gomock_generator/gomockgenerator/generate.go:98 +0xd4                                                       
created by github.com/Scalingo/go-utils/gomock_generator/gomockgenerator.GenerateMocks
        /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/go-utils/gomock_generator/gomockgenerator/generate.go:96 +0x89e                                                      
```

--- a/gomock_generator/gomockgenerator/generate.go
+++ b/gomock_generator/gomockgenerator/generate.go
@@ -98,6 +98,7 @@ func GenerateMocks(ctx context.Context, gcfg GenerationConfiguration, mocksCfg M
 			path, sig, err := generateMock(ctx, mocksCfg.BasePackage, mock, mockSigs)
 			if err != nil {
 				log.Error(err)
+				wg.Done()
 				return
 			}
 			lock.Lock()

--- a/gomock_generator/gomockgenerator/generate.go
+++ b/gomock_generator/gomockgenerator/generate.go
@@ -94,18 +94,19 @@ func GenerateMocks(ctx context.Context, gcfg GenerationConfiguration, mocksCfg M
 	for _, mock := range mocksCfg.Mocks {
 		wg.Add(1)
 		go func(mock MockConfiguration) {
+			defer func() {
+				wg.Done()
+				<-sem
+			}()
 			sem <- true
 			path, sig, err := generateMock(ctx, mocksCfg.BasePackage, mock, mockSigs)
 			if err != nil {
 				log.Error(err)
-				wg.Done()
 				return
 			}
 			lock.Lock()
 			newMockSigs[path] = sig
 			lock.Unlock()
-			<-sem
-			wg.Done()
 		}(mock)
 	}
 	wg.Wait()

--- a/gomock_generator/gomockgenerator/package.go
+++ b/gomock_generator/gomockgenerator/package.go
@@ -51,24 +51,31 @@ func interfaceSignature(pkg, iName string) (string, error) {
 									if typeSpec.Name.String() == iName {
 										var interfaceSig string
 										for _, m := range interfaceType.Methods.List {
-											methodName := m.Names[0].String()
-											funcType := m.Type.(*ast.FuncType)
-											var methodType string
-
-											if funcType.Results != nil {
-												for _, res := range funcType.Results.List {
-													methodType = fmt.Sprintf("%s,%s", methodType, fieldToString(res.Type))
+											switch v := m.Type.(type) {
+											case *ast.Ident:
+												interfaceSig = fmt.Sprintf("%s\nInterface{%s}\n", interfaceSig, v.Name)
+											case *ast.FuncType:
+												methodName := m.Names[0].String()
+												var methodType string
+												if v.Results != nil {
+													for _, res := range v.Results.List {
+														methodType = fmt.Sprintf("%s,%s", methodType, fieldToString(res.Type))
+													}
 												}
-											}
 
-											var methodParams string
-											if funcType.Params != nil {
-												for _, param := range funcType.Params.List {
-													methodParams = fmt.Sprintf("%s,%s", methodParams, fieldToString(param.Type))
+												var methodParams string
+												if v.Params != nil {
+													for _, param := range v.Params.List {
+														methodParams = fmt.Sprintf("%s,%s", methodParams, fieldToString(param.Type))
+													}
 												}
+												interfaceSig = fmt.Sprintf("%s\n%s(%s)(%s)\n", interfaceSig, methodName, methodParams, methodType)
+
+											default:
+												panic("Unexpected AST type")
 											}
-											interfaceSig = fmt.Sprintf("%s\n%s(%s)(%s)\n", interfaceSig, methodName, methodParams, methodType)
 										}
+										fmt.Println(interfaceSig)
 										return interfaceSig, nil
 									}
 								}

